opencti-platform/opencti-graphql/src/database/ai-llm.ts

```diff
@@ -22,6 +22,29 @@ const AI_MODEL = conf.get('ai:model');
 const AI_MAX_TOKENS = conf.get('ai:max_tokens');
 const AI_VERSION = conf.get('ai:version');
 
+/**
+ * Extracts the Azure instance name and deployment name from a URL of the form:
+ * https://<my-instance-name>.openai.azure.com/openai/deployments/<my-deployment-name>/
+ *
+ * @param {string} url - The full URL containing the Azure instance and deployment names.
+ * @returns {[string, string] | null} - Returns a tuple [azureInstance, azureDeployment] or null if not matched.
+ */
+export function extractAzureInstanceAndDeployment(
+  url: string,
+): [string, string] | null {
+  const regex = /^https:\/\/([^.]+)\.openai\.azure\.com\/openai\/deployments\/([^/]+)\/*/;
+  const match = url.match(regex);
+
+  if (!match) {
+    return null;
+  }
+
+  const [, azureInstance, azureDeployment] = match;
```

Comment:
```suggestion
  const { azureInstance, azureDeployment } = match.groups;
```
if you implement the name extraction in previous, I find it more readable

Reply:
Named capturing groups are only available when targeting 'ES2018' or later.

We need to modify the tsconfig.json with eg. "target": "ES2020" or later